inc/saiport.h

```diff
@@ -388,6 +388,136 @@ typedef enum _sai_port_connector_failover_mode_t
+ * Used for MDIX mode configuration
+ */
+typedef enum _sai_port_mdix_mode_config_t
+{
+    /** MDIX mode status auto */
+    SAI_PORT_MDIX_MODE_CONFIG_AUTO,
+
+    /** MDIX mode status straight */
+    SAI_PORT_MDIX_MODE_CONFIG_STRAIGHT,
+
+    /**  MDIX mode status cross over */
+    SAI_PORT_MDIX_MODE_CONFIG_CROSSOVER
+} sai_port_mdix_mode_config_t;
+
+/**
+ * @brief Attribute data for #SAI_PORT_ATTR_AUTO_NEG_CONFIG_MODE
+ * Used for auto negotiation mode to configure master or slave mode
+ */
+typedef enum _sai_port_auto_neg_config_mode_t
+{
+    /** Auto neg mode auto */
+    SAI_PORT_AUTO_NEG_CONFIG_MODE_AUTO,
+
+    /** Auto neg mode slave */
+    SAI_PORT_AUTO_NEG_CONFIG_MODE_SLAVE,
+
+    /** Auto neg mode master */
+    SAI_PORT_AUTO_NEG_CONFIG_MODE_MASTER
+} sai_port_auto_neg_config_mode_t;
+
+/**
+ * @brief Attribute data for #SAI_PORT_ATTR_MODULE_TYPE
+ * Used for configuring Fiber module type
+ */
+typedef enum _sai_port_module_type_t
+{
+    /** Module Type Fiber */
+    SAI_PORT_MODULE_TYPE_1000BASE_X,
+
+    /** Module Type 100FX */
+    SAI_PORT_MODULE_TYPE_100FX,
+
+    /** Module Type SGMII-Slave */
+    SAI_PORT_MODULE_TYPE_SGMII_SLAVE,
+} sai_port_module_type_t;
+
+/**
+ * @brief Attribute data for #SAI_PORT_ATTR_DUAL_MEDIA
+ * Used to configure media type for dual media supported PHY
+ */
+typedef enum _sai_port_dual_media_t
+{
+    /**  Dual media not supported */
+    SAI_PORT_DUAL_MEDIA_NONE,
+
+    /**  Force Copper mode, Fiber is inactive/disabled */
+    SAI_PORT_DUAL_MEDIA_COPPER_ONLY,
+
+    /**  Force Fiber mode, Copper is inactive/disabled */
+    SAI_PORT_DUAL_MEDIA_FIBER_ONLY,
+
+    /**  Both Copper and Fiber supported, but Copper preferred */
+    SAI_PORT_DUAL_MEDIA_COPPER_PREFERRED,
+
+    /**  Both Copper and Fiber supported, but Fiber preferred */
+    SAI_PORT_DUAL_MEDIA_FIBER_PREFERRED
+} sai_port_dual_media_t;
+
+/**
+ * @brief Attribute data for #SAI_PORT_ATTR_CABLE_PAIR_STATE
+ * Copper cable pair states
+ */
+typedef enum _sai_port_cable_pair_state_t
+{
+    /**  Cable state no faults */
+    SAI_PORT_CABLE_PAIR_STATE_OK,
+
+    /**  Cable state open */
+    SAI_PORT_CABLE_PAIR_STATE_OPEN,
+
+    /**  Cable state short */
+    SAI_PORT_CABLE_PAIR_STATE_SHORT,
+
+    /**  Cable state cross talk */
+    SAI_PORT_CABLE_PAIR_STATE_CROSSTALK,
+
+    /**  Cable state unknown */
+    SAI_PORT_CABLE_PAIR_STATE_UNKNOWN
+} sai_port_cable_pair_state_t;
+
+/**
+ * @brief Attribute data for #SAI_PORT_ATTR_CABLE_TYPE
+ * Copper cable types
+ */
+typedef enum _sai_port_cable_type_t
+{
+    /**  Cable type Unknown */
+    SAI_PORT_CABLE_TYPE_UNKNOWN,
+
+    /**  Cable type CAT5 */
+    SAI_PORT_CABLE_TYPE_CAT5,
+
+    /**  Cable type CAT5E */
+    SAI_PORT_CABLE_TYPE_CAT5E,
+
+    /**  Cable type CAT6 */
+    SAI_PORT_CABLE_TYPE_CAT6,
+
+    /**  Cable type CAT6A */
+    SAI_PORT_CABLE_TYPE_CAT6A,
+
+    /**  Cable type CAT7 */
+    SAI_PORT_CABLE_TYPE_CAT7,
+} sai_port_cable_type_t;
```

Comment:
In the case of legacy Broadcom PHY, the cable-length is only valid when the state is OK, and by having state and cable-length tested in 2 separated calls could lead to a race condition, maybe we should deal with them in one single call, agree?

Reply:
No. I don't think we can do it in single call. I was thinking if we should add a @validonly for the cable length but then we can't support the other PHYs which do not require, cable state to be OK to get the cable length. I think user can check the cable state before querying the cable length on the legacy PHYs.